spring-pulsar-test/src/test/java/org/springframework/pulsar/test/support/ConditionsTest.java

```diff
@@ -0,0 +1,151 @@
+
+			boolean result = consumedMessagesCondition.meets(List.of(
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-0".getBytes()), Schema.STRING,
+							"topic"),
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-1".getBytes()), Schema.STRING,
+							"topic")));
+
+			Assertions.assertThat(result).isTrue();
+		}
+
+		@Test
+		void receivedMessageCountDoesNotMeetCondition() {
+			ConsumedMessagesCondition<String> consumedMessagesCondition = desiredMessageCount(3);
+
+			boolean result = consumedMessagesCondition.meets(List.of(
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-0".getBytes()), Schema.STRING,
+							"topic"),
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-1".getBytes()), Schema.STRING,
+							"topic")));
+
+			Assertions.assertThat(result).isFalse();
+		}
+
+		@ParameterizedTest
+		@ValueSource(ints = { 0, -1 })
+		void throwExceptionWhenDesiredMessageCountEqualOrLessThanZero(int messageCount) {
+			Assertions.assertThatThrownBy(() -> desiredMessageCount(messageCount))
+				.isInstanceOf(IllegalStateException.class)
+				.hasMessage("Desired message count must be greater than 0");
+		}
+
+	}
+
+	@Nested
+	class ContainsValueConsumedMessagesCondition {
+
+		@Test
+		void messageValueContainsExpectation() {
+			ConsumedMessagesCondition<String> consumedMessagesCondition = Conditions.containsValue("message-1");
+
+			boolean result = consumedMessagesCondition.meets(List.of(
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-0".getBytes()), Schema.STRING,
+							"topic"),
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-1".getBytes()), Schema.STRING,
+							"topic"),
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-2".getBytes()), Schema.STRING,
+							"topic")));
+
+			Assertions.assertThat(result).isTrue();
+		}
+
+		@Test
+		void messageValueDoesNotContainExpectation() {
+			ConsumedMessagesCondition<String> consumedMessagesCondition = Conditions.containsValue("message-3");
+
+			boolean result = consumedMessagesCondition.meets(List.of(
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-0".getBytes()), Schema.STRING,
+							"topic"),
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-1".getBytes()), Schema.STRING,
+							"topic"),
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-2".getBytes()), Schema.STRING,
+							"topic")));
+
+			Assertions.assertThat(result).isFalse();
+		}
+
+	}
+
+	@Nested
+	class ContainsValuesConsumedMessagesCondition {
+
+		@Test
+		void messageValueContainsExpectation() {
+			ConsumedMessagesCondition<String> consumedMessagesCondition = Conditions.containsValues("message-1",
+					"message-2");
+
+			boolean result = consumedMessagesCondition.meets(List.of(
+					MessageImpl.create(new MessageMetadata(), ByteBuffer.wrap("message-0".getBytes()), Schema.STRING,
```

Comment:
I think creating a convenience method that takes a string and returns a `MessageImpl` would reduce these tests quite a bit.